pkgs/applications/graphics/structorizer/default.nix

```diff
@@ -0,0 +1,84 @@
+{ stdenv, lib, fetchFromGitHub, jdk11, makeDesktopItem, makeWrapper
+, copyDesktopItems }:
+
+stdenv.mkDerivation rec {
+  pname = "structorizer";
+  version = "3.32-11";
+
+  desktopItems = [
+    (makeDesktopItem {
+      type = "Application";
+      name = "Structorizer";
+      desktopName = "Structorizer";
+      genericName = "Diagram creator";
+      comment = meta.description;
+      icon = "structorizer";
+      exec = "structorizer";
+      terminal = false;
+      categories = [
+        "Development"
+        "Graphics"
+        "VectorGraphics"
+        "RasterGraphics"
+        "ComputerScience"
+      ];
+      keywords = [ "nsd" "diagrams" ];
+    })
+  ];
+
+  src = fetchFromGitHub {
+    owner = "fesch";
+    repo = "Structorizer.Desktop";
+    rev = version;
+    hash = "sha256-rGyeOcGm6uBplgTjMIOy/xRekfHacwDy9kkMigmRSdk=";
+  };
+
+  nativeBuildInputs = [ jdk11 makeWrapper copyDesktopItems ];
+
+  patches = [ ./makeStructorizer.patch ./makeBigJar.patch ];
+
+  postPatch = ''
+    chmod +x makeStructorizer
+    chmod +x makeBigJar
+
+    patchShebangs --build makeStructorizer
+    patchShebangs --build makeBigJar
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    ./makeStructorizer
+    ./makeBigJar
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    install -D ${pname}.jar -t $out/share/java/
+    install -d $out/bin
+    makeWrapper ${jdk11}/bin/java $out/bin/${pname} \
+      --add-flags "-jar $out/share/java/${pname}.jar"
+
+    cd src/lu/fisch/structorizer/gui
+    install -D icons/000_structorizer.png $out/share/icons/hicolor/16x16/apps/structorizer.png
+    install -D icons_24/000_structorizer.png $out/share/icons/hicolor/24x24/apps/structorizer.png
+    install -D icons_32/000_structorizer.png $out/share/icons/hicolor/32x32/apps/structorizer.png
+    install -D icons_48/000_structorizer.png $out/share/icons/hicolor/48x48/apps/structorizer.png
+    install -D icons_64/000_structorizer.png $out/share/icons/hicolor/64x64/apps/structorizer.png
+    install -D icons_128/000_structorizer.png $out/share/icons/hicolor/128x128/apps/structorizer.png
+    install -D icons_256/000_structorizer.png $out/share/icons/hicolor/256x256/apps/structorizer.png
```

Comment:
you could consider doing this in a for loop, i'm not sure whether it's worth it though.

Reply:
I think it's worth it, pushed it.